### ISSUE TITLE
`soroban network start` followup improvements

### ISSUE THREAD

**Issue by elizabethengelman** [stellar/stellar-cli#1230]
The `soroban network start` and `soroban network stop` commands were added in PR # 1107, and as part of the PR discussion there were some follow up things that we wanted to improve on:

## Todo
- [x] Add loading animation and some more information about what is happening as the docker container starts up (see https://github.com/stellar/soroban-cli/pull/1107/#discussion_r1456367830) - **we decided that mentioning to the user that `network container logs` exists is enough**
- [x] Try several runtime defaults in parallel and return when one of them connects, see (https://github.com/stellar/soroban-cli/pull/1107/#discussion_r1490271348) - **we are already trying the most common default ("unix:///var/run/docker.sock" for UNIX & mac, and "npipe:////./pipe/docker_engine" for windows). if that doesn't work, which is common with docker desktop, we try "home/.docker/run/docker.sock". If that doesn't work, a user is still able to override the defaults with the `docker_host` flag**
- [x] Does `soroban network start` make sense? Or should this eventually move to this to `soroban instance start | stop` or `soroban network instance start | stop`? See https://github.com/stellar/soroban-cli/pull/1107#discussion_r1491653919 for more info -> _this was changed to soroban network container start_
- [x] Better error handling when the container doesn't exist (specifically with the `stop` command)
- [x] Add a flag to change the name passed to docker

## On the chopping block:
- [ ] Add a healthcheck that makes sure that the docker container is up and running before releasing the user back to the command line (see https://github.com/stellar/soroban-cli/pull/1107/#discussion_r1456367830) - _**Instead of a healthcheck, I've added the first few lines of the conatiner's logs into the command output, and also included a note about how to use `stellar network container logs` to see the rest of the logs. Is this good, or do we think that a health check is important?**_
- [ ] support SSL connections - _**I'm thinking that we could potentially skip this task. Since the quickstart is meant to be used for testing and development purposes, it seems okay to not have the connections be encrypted. created a [new issue](https://github.com/stellar/stellar-cli/issues/1429) in case we run into this issue integration with laboratory**

**Comment by leighmcculloch** [comment]
> * [ ]  Does `soroban network start` make sense? Or should this eventually move to this to `soroban instance start | stop` or `soroban network instance start | stop`? See [feat: start network docker container with cli #1107 (comment)](https://github.com/stellar/soroban-cli/pull/1107#discussion_r1491653919) for more info

I think `soroban instance start` will serve us better, and be less confusing long term. Main motivation for separating it being that by bundling it into `network` we're making `network` do two mostly unrelated things:
1. Configuring connecting to networks.
2. Running networks / services.

While these things both involve networks, they're otherwise unrelated bundles of operations.

For example, take the following help output. It is easy to think that `rm` and `stop` both operate on local running networks, but actually `rm` removes configurations for the client. It is also easy to think that `add` might create a new network while `start` will boot it up, but actually `add` adds a configuration for the client and start is what creates a new network.

```
❯ soroban network -h
Configure networks

Usage: soroban network <COMMAND>

Commands:
  add  Add a new network
  rm   Remove a network
  ls   List networks
```

I think the relationship between the subcommands is much clearer if separated:

```
❯ soroban network -h
Configure networks for the client to connect to.

Usage: soroban instance <COMMAND>

Commands:
  add  Add a new network
  rm   Remove a network
  ls   List networks

❯ soroban instance -h
Run local instances of the Stellar network, RPC, and Horizon for local development and testing, as well as for joining testnet and pubnet networks.

Usage: soroban instance <COMMAND>

Commands:
  start Start a network
  stop Stop a network
```

**Comment by leighmcculloch** [comment]
Suggested follow up:

- [ ] Auto-configure the network client with a local network for the running instance.

**Comment by janewang** [comment]
Add stellar network container logs command in the terminal and can close
